akka-http-core/src/test/scala/akka/http/scaladsl/model/HttpMethodSpec.scala

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (C) 2009-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.model
+
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+class HttpMethodSpec extends AnyWordSpec with Matchers {
+  "HttpMethod.custom()" should {
+    "accept a valid name" in {
+      HttpMethod.custom("Yes.Thi$_is_1~'VAL|D`_me+h*d-^ame!#%&")
```

Comment:
The most reasonable custom method name I ever saw :D

Reply:
It's specified in RFC666 - Gaslighting HTTP resources.